src/utilities/tableHelpers.ts

```diff
@@ -0,0 +1,21 @@
+import type { TablePreferences, TablePaginationAttributes } from '@/types'
+
+export const DEFAULT_PAGE_SIZE = 15
+
+/**
+ * @description Get the initial page size for the table
+ * @param {TablePreferences} tablePreferences
+ * @param {TablePaginationAttributes} paginationAttributes
+ * @returns {number} The initial page size
+ */
+export const getInitialPageSize = (tablePreferences: TablePreferences, paginationAttributes: TablePaginationAttributes): number => {
+  if (tablePreferences.pageSize) {
+    return tablePreferences.pageSize
+  } else if (paginationAttributes.initialPageSize) {
+    return paginationAttributes.initialPageSize
+  } else if (paginationAttributes.pageSizes) {
```

Comment:
issue: I think you need to check that it's an array?
```suggestion
  } else if (typeof paginationAttributes.pageSizes === 'object' && paginationAttributes.pageSizes.length) {
```

Reply:
Checking type isn't really necessary since if something other than array of numbers was passed through `paginationAttributes` prop Vue would throw a type error, however checking array length is useful.

Reply:
If you passed `paginationAttributes.pageSizes = 'my-name'` you would get a length of `7`

You'd get prop validation _warnings_ in dev mode, but not necessarily errors in a consuming app, depending on the setup

Reply:
Added a check to make sure `paginationAttributes.pageSizes` is an array